src/Test/Hspec/Hedgehog.hs

```diff
@@ -184,7 +184,9 @@ instance (m ~ IO) => Example (a -> PropertyT m ()) where
                             NoConfidenceTermination (TestLimit _) ->
                                 NoConfidenceTermination (TestLimit maxTests)
                     , propertyDiscardLimit =
-                        DiscardLimit $ maxDiscardRatio qcArgs
+                        -- QuickCheck uses a discard ratio, while hedgehog uses
+                        -- an absolute limit.
+                        DiscardLimit $ maxDiscardRatio qcArgs * maxTests
```

Comment:
Code comments are a code smell, as they tend to go out of sync with the actual implementation.

For that reason:

1. If you think it's important, you can introduce a local variable that describes in code what your comment is trying to convey, e.g. `absoluteDiscardLimit = maxDiscardRatio qcArgs * maxTests`.
1. Alternatively, you may just delete the comment.